### PR TITLE
tests: improve test suite

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Use `toBe()` matcher for primitives.

### CODE REVIEW

**Review by liyasthomas — APPROVED** [review]

**Comment by AndrewBastin** [comment]
> Use `toBe()` matcher for primitives.

About the `toBe()` usage. Yes, for primitives it is fine, but I want tests to be consistent and its behaviour easily predictable. `toBe()` checks for referential equality in non-primitive cases which is a really obscure behaviour.

I want to move the development process to a more TDD style and expect tests to be written for all the new code and I believe `toBe()` adds a wee bit of confusion to people on its behaviour.

Hence, I want to reserve `toBe()` just for referential equality checks.

@liyasthomas

**Comment by liyasthomas** [comment]
I guess `toBe()` suits best for primitives like strings, numbers and booleans. For objects, and everything else I prefer `toEqual()`. Wdy say @jamesgeorge007 ?

**Comment by AndrewBastin** [comment]
@liyasthomas that is what James prefers too, but I am suggesting to keep `toEqual()` for primitives as well.

**Comment by liyasthomas** [comment]
Okay, I second Andrew if somehow `toBe()` cause any confusion for a new user.

File: components/ui/__tests__/autocomplete.spec.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ const props = {
   source: ["app", "apple", "appliance", "brian", "bob", "alice"],
 }
 
+// ["pp", "pple", "ppliance", "lice"]
+const suggestionStr = props.source.filter((str) => str.startsWith("a")).map((str) => str.slice(1))
```
Comment (jamesgeorge007) [inline components/ui/__tests__/autocomplete.spec.js:12]: Should we keep this change?

Comment (AndrewBastin) [reply components/ui/__tests__/autocomplete.spec.js:12]: Yup, this is fine.

**Pull request merged**